### ISSUE TITLE
[Feature]: Enable iOS Silent Push Notifications in Capacitor

### ISSUE THREAD

**Issue by bluepuma77** [ionic-team/capacitor-plugins#2038]
### Description

Support iOS Silent Push Notifications to be able to trigger JavaScript functions within Capacitor

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web

### Request or proposed solution

Support iOS Silent Push Notifications to be able to trigger JavaScript functions like a file download. Use cases are many, you may need to update content/files within the app, and it needs to be available next time the app is opened, even when no network is available, like on most flights.

### Alternatives

The [doc](https://capacitorjs.com/docs/apis/push-notifications#silent-push-notifications--data-only-notifications) states to use native implementation, but this kind of defeats the purpose of Capacitor.

### Additional Information

I'm hoping this can be easily implemented with a few lines of code as the regular push notifications are already working.

**Comment by bluepuma77** [comment]
Thanks @jcesarmobile for moving this to the right place. 

Can you remove the tags `android` and `web`? It seems I can not change it.

**Comment by aparajita** [comment]
We have a simple plugin we have been using that implements this. I'll publish it to npm next week.

**Comment by bluepuma77** [comment]
@aparajita That would be great! 🥳

**Comment by bluepuma77** [comment]
What's the actual problem with `@capacitor/push-notifications` library, that it is not compatible with "iOS Silent Push Notifications"? Shouldn't it really just need very few lines of code to make it work? Happy to support testing.

@markemer Do you have any insights? It seems you were the last one working on this module.

**Comment by aparajita** [comment]
@bluepuma77 yes it would require only a few lines of code, at least that has been our experience. Not sure why they didn't implement it.

**Comment by bluepuma77** [comment]
@aparajita You offered to create a npm package. To make it easier for you, could you simply provide a template on Github?

**Comment by aparajita** [comment]
Here's what we are using currently. Needs a little updating, but works: https://github.com/willsub/capacitor-ios-remote-notifications

**Comment by bluepuma77** [comment]
Found two more related Github repos:

1. [capacitor-ios-background-push](https://github.com/Exxili/capacitor-ios-background-push) would not compile, opened an [issue](https://github.com/Exxili/capacitor-ios-background-push/issues/1)
2. [capacitor-plugin-silent-notifications](https://github.com/greenygh0st/capacitor-plugin-silent-notifications) finally seems to work, sending `SilentNotification` with [apns2](https://github.com/AndrewBarba/apns2)

**Comment by aparajita** [comment]
@bluepuma77 I just released [capacitor-ios-silent-notifications](https://github.com/aparajita/capacitor-ios-silent-notifications).

**Comment by Eraldo** [comment]
> @bluepuma77 I just released [capacitor-ios-silent-notifications](https://github.com/aparajita/capacitor-ios-silent-notifications).

Thanks @aparajita. 🙏 
PS: Would be nice to have this integrated into `@capacitor/push-notifications`. 😅 
You did a nice job of showing how it's done. 👏

**Comment by eric-horodyski** [comment]
This thread has been inactive for some time, and we’re cleaning up older items to keep the backlog focused and manageable.

Details on the effort:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still relevant today, feel free to open a new issue with current information and we’ll review it.
